src/coreclr/vm/codeman.cpp

```diff
@@ -2428,7 +2444,15 @@ HeapList* LoaderCodeHeap::CreateCodeHeap(CodeHeapRequestInfo *pInfo, LoaderHeap
     HeapList *pHp = new HeapList;
 
 #if defined(TARGET_AMD64) || defined(TARGET_ARM64) || defined(TARGET_LOONGARCH64) || defined(TARGET_RISCV64)
-    pHp->CLRPersonalityRoutine = (BYTE *)pCodeHeap->m_LoaderHeap.AllocMem(JUMP_ALLOCATE_SIZE);
+    if (pInfo->IsInterpreted())
+    {
+        pHp->CLRPersonalityRoutine = NULL;
+        pCodeHeap->m_LoaderHeap.ReservePages(1);
```

Comment:
Is this important? I assume the idea is to ensure that the heap has something in it?

Reply:
Yes, the code further requires at least one reserved page to be present. For the personality routine case in the else branch, the `AllocMem` ensures that (we alloc hence a page gets reserved due to that).  
There is a comment in the #else below explaining it for the case of targets that don't use personality routines.